graphql-bal-service/src/main/java/io/ballerina/graphql/schema/generator/SdlSchemaGenerator.java

```diff
@@ -0,0 +1,238 @@
+import io.ballerina.graphql.schema.diagnostic.DiagnosticMessages;
+import io.ballerina.graphql.schema.exception.SchemaGenerationException;
+import io.ballerina.projects.DiagnosticResult;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.ModuleId;
+import io.ballerina.projects.Package;
+import io.ballerina.projects.PackageCompilation;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.ProjectKind;
+import io.ballerina.projects.directory.ProjectLoader;
+import io.ballerina.stdlib.graphql.commons.types.Schema;
+import io.ballerina.stdlib.graphql.commons.utils.SdlSchemaStringGenerator;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.graphql.schema.Constants.EMPTY_STRING;
+import static io.ballerina.graphql.schema.Constants.PERIOD;
+import static io.ballerina.graphql.schema.utils.Utils.createOutputDirectory;
+import static io.ballerina.graphql.schema.utils.Utils.formatBasePath;
+import static io.ballerina.graphql.schema.utils.Utils.getDecodedSchema;
+import static io.ballerina.graphql.schema.utils.Utils.getSchemaString;
+import static io.ballerina.graphql.schema.utils.Utils.getSdlFileName;
+import static io.ballerina.graphql.schema.utils.Utils.getServiceBasePath;
+import static io.ballerina.graphql.schema.utils.Utils.isGraphqlService;
+import static io.ballerina.graphql.schema.utils.Utils.resolveSchemaFileName;
+import static io.ballerina.graphql.schema.utils.Utils.writeFile;
+import static io.ballerina.stdlib.graphql.commons.utils.Utils.isGraphQLServiceObjectDeclaration;
+
+/**
+ * This class implements the GraphQL SDL schema generation.
+ */
+public class SdlSchemaGenerator {
+
+    private static PrintStream outStream = System.out;
+
+    /**
+     * Export the SDL schema for given Ballerina GraphQL services.
+     */
+    public static void generate(Path filePath, Path outPath, String serviceName)
+            throws SchemaGenerationException {
+        Project project = ProjectLoader.loadProject(filePath);
+        PackageCompilation compilation = getPackageCompilation(project);
+        Package packageName = project.currentPackage();
+        DocumentId docId;
+        Document doc;
+        if (project.kind().equals(ProjectKind.BUILD_PROJECT)) {
+            docId = project.documentId(filePath);
+            ModuleId moduleId = docId.moduleId();
+            doc = project.currentPackage().module(moduleId).document(docId);
+        } else {
+            Module currentModule = packageName.getDefaultModule();
+            Iterator<DocumentId> documentIterator = currentModule.documentIds().iterator();
+            docId = documentIterator.next();
+            doc = currentModule.document(docId);
```

Comment:
This always picks the first item of the `documentIds`. Is that intended? 

If that's the case, why do we need the iterator?

Reply:
Removed the iterator.